src/components/ColorSwatches/index.js

```diff
@@ -0,0 +1,144 @@
+import React, { useState, useEffect } from 'react';
+import styled from 'styled-components';
+import Button from '../../components/Button';
+
+function ColorSwatches({
+  currentColor,
+  setDesktopState,
+  setOpenColorSwatches,
+}) {
+  const [customColor, setCustomColor] = useState(false);
+  const [highlighted, setHighlighted] = useState(currentColor);
+
+  const sampleColumn1 = ['#ffffff', '#fa0000', '#00fc00', '#0602fd', '#bddbc3'];
+  const sampleColumn2 = ['#000000', '#7c0001', '#027f00', '#040078', '#a3c9ee'];
+  const sampleColumn3 = ['#bfbfc1', '#fcfb09', '#00fcfd', '#fd06f5', '#fcf9f0'];
+  const sampleColumn4 = ['#818181', '#7f7e09', '#03807a', '#641c5c', '#a2a1a7'];
+
+  const colorsArray = [
+    sampleColumn1,
+    sampleColumn2,
+    sampleColumn3,
+    sampleColumn4,
+  ];
+
+  useEffect(() => {
+    const isSystemColor = checkIfSystemColor(currentColor, colorsArray);
+    if (isSystemColor) {
+      setCustomColor(false);
+    } else {
+      setCustomColor(currentColor);
+    }
+  }, [currentColor]);
+
+  const checkIfSystemColor = (colorToCheck, systemColors) => {
+    let existSystemColor = false;
```

Comment:
Not the best name here; I would suggest either `isSystemColor` (which possible sounds like a function) or something like `found`.

💡 If you would have gone with a single array, a simple `systemColors.includes` would suffice with no need for this helper. Another route could be to `flat` the outer array and then run the `includes` on that - But could also be an overkill.